### PR TITLE
Fix: resolve locally installed addons not being displayed

### PR DESCRIPTION
**Pull request by charlie0129**
Signed-off-by: Charlie Chiang <charlie_c_0129@outlook.com>

### Description of your changes

Addressed an issue where locally installed addons being displayed as the registry one if there is an addon with the same name is in the registry.

#### Before

Notice the `dex` addon, which is installed from local sources, but displayed as the one in the registry.

Screen Shot 2022-05-07 at 7 40 04 PM

#### After

It is now displayed properly as locally installed (the one in  KubeVela registry is not affected).

Screen Shot 2022-05-07 at 7 41 52 PM

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Changes that need to apply to existing unit tests have been added.

### CODE REVIEW

File: references/cli/addon.go [hunk 1]
```diff
@@ -480,7 +480,25 @@ func listAddons(ctx context.Context, clt client.Client, registry string) error {
...
+		labels := app.GetLabels()
+		addonName := labels[oam.LabelAddonName]
+		addonVersion := labels[oam.LabelAddonVersion]
+		table.AddRow(addonName, app.GetLabels()[oam.LabelAddonRegistry], "", genAvailableVersionInfo([]string{addonVersion}, addonVersion), statusEnabled)
+		onlineAddon[addonName] = true
+	}
+
 	for _, addon := range addons {
+		// if the addon with same name has already installed locally, display the registry one as not installed
+		if onlineAddon[addon.Name] {
```
Comment (wangyikewxgm) [inline references/cli/addon.go:498]: Change the map name `onlineAddon` to `InstalledLocallyAddon`?

**Review by wonderflow — COMMENTED** [review]
good job, please add some test cases

**Comment by charlie0129** [comment]
Thank you for you suggestions. Relevant tests have been added.

**Review by wonderflow — APPROVED** [review]
good job

**Review by Somefive — APPROVED** [review]

**Pull request merged**